components/support/nimbus-fml/src/backends/swift/gen_structs/bundled.rs

```diff
@@ -0,0 +1,155 @@
+    /// N.B. `Literal` is aliased from `serde_json::Value`.
+    fn literal(
+        &self,
+        _oracle: &dyn CodeOracle,
+        _ctx: &dyn Display,
+        _renderer: &dyn LiteralRenderer,
+        literal: &Literal,
+    ) -> String {
+        match literal {
+            serde_json::Value::String(v) => quoted(v),
+            _ => unreachable!("Expecting a string"),
+        }
+    }
+
+    fn defaults_type(&self, oracle: &dyn CodeOracle) -> String {
+        oracle.find(&TypeIdentifier::String).type_label(oracle)
+    }
+
+    fn defaults_mapper(
+        &self,
+        _oracle: &dyn CodeOracle,
+        value: &dyn Display,
+        vars: &dyn Display,
+    ) -> Option<String> {
+        Some(format!(
+            "{vars}.resourceBundles.getString(named: {value}) ?? {value}",
+            vars = vars,
+            value = value
+        ))
+    }
+}
+
+pub(crate) struct ImageCodeType;
+
+impl CodeType for ImageCodeType {
+    /// The language specific label used to reference this type. This will be used in
+    /// method signatures and property declarations.
+    fn type_label(&self, _oracle: &dyn CodeOracle) -> String {
+        "UIImage".into()
+    }
+
+    fn property_getter(
+        &self,
+        oracle: &dyn CodeOracle,
+        vars: &dyn Display,
+        prop: &dyn Display,
+        default: &dyn Display,
+    ) -> String {
+        code_type::property_getter(self, oracle, vars, prop, default)
+    }
+
+    fn value_getter(
+        &self,
+        oracle: &dyn CodeOracle,
+        vars: &dyn Display,
+        prop: &dyn Display,
+    ) -> String {
+        code_type::value_getter(self, oracle, vars, prop)
+    }
+
+    fn value_mapper(&self, oracle: &dyn CodeOracle) -> Option<String> {
+        code_type::value_mapper(self, oracle)
+    }
+
+    /// The name of the type as it's represented in the `Variables` object.
+    /// The string return may be used to combine with an indentifier, e.g. a `Variables` method name.
+    fn variables_type(&self, _oracle: &dyn CodeOracle) -> VariablesType {
+        VariablesType::Image
+    }
+
+    /// A representation of the given literal for this type.
+    /// N.B. `Literal` is aliased from `serde_json::Value`.
+    fn literal(
+        &self,
+        _oracle: &dyn CodeOracle,
+        _ctx: &dyn Display,
+        _renderer: &dyn LiteralRenderer,
+        literal: &Literal,
+    ) -> String {
+        match literal {
+            serde_json::Value::String(v) => quoted(v),
+            _ => unreachable!("Expecting a string matching an image/drawable resource"),
+        }
+    }
+
+    fn defaults_type(&self, oracle: &dyn CodeOracle) -> String {
+        oracle.find(&TypeIdentifier::String).type_label(oracle)
+    }
+
+    fn defaults_mapper(
+        &self,
+        _oracle: &dyn CodeOracle,
+        value: &dyn Display,
+        vars: &dyn Display,
+    ) -> Option<String> {
+        Some(format!(
+            "{vars}.resourceBundles.getImage(named: {value})!",
```

Comment:
So my assumption here (I should've commented this): this is _only_ for message defaults. In Android this would be a compile time error if the image isn't in the bundle.

On iOS this would be a crash in development.

Reply:
I added a `getImageNotNull` method which fatalErrors with a nice error message, instead of opaquely force dereferencing.